10_rest/alextmz/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/anz-bank/go-course/10_rest/alextmz/pkg/puppy"
+	"github.com/anz-bank/go-course/10_rest/alextmz/pkg/puppy/store"
+	"github.com/anz-bank/go-course/10_rest/alextmz/pkg/rest"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	args      = os.Args[1:]
+	flagfile  = kingpin.Flag("data", "JSON file to read").Short('d').Default("./test/valid-formatted-json.json").String()
+	flagport  = kingpin.Flag("port", "TCP port to listen on").Short('p').Default("7735").Uint16()
+	flagstore = kingpin.Flag("store", "Backing store to use").Short('s').Default("sync").Enum("map", "sync")
+
+	out io.Writer = os.Stdout
+
+	// shutdownhttp signals main() to... shutdown the http server
+	shutdownhttp = make(chan bool)
```

Comment:
My intent was to do real connection testing on `main()`, not just handler testing, and this was a (relatively) elegant way I found to `http.Shutdown` a `http.ListenAndServe`; but as I later found out, Travis doesn't allow connections to `localhost`. 
So this could had been substituted for a global `http.Server` variable, and the test could had `http.Shutdown` the server it before calling `main()`. 
Which also doesn't feel a good solution despite the simplicity.
So I left it in as it works, and doesn't use another global.
I hope reviewers like it but I admit I'm not high on hopes 🤞